src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -1306,7 +1313,44 @@ public virtual string Labels
         public virtual string ProxyUriPath { get { return _localConfiguration.service.proxy.uriPath; } }
         public virtual int ProxyPort { get { return _localConfiguration.service.proxy.port; } }
         public virtual string ProxyUsername { get { return _localConfiguration.service.proxy.user; } }
-        public virtual string ProxyPassword { get { return _localConfiguration.service.proxy.password; } }
+
+        private string _obscuringKey;
+        public string ObscuringKey
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_obscuringKey))
```

Comment:
Need to add check variable here.  Otherwise, if an obscuring key is not supplied, this getter will evaluate every time.

One could say that we are only using the ObscuringKey in one place, so this would only happen in one place.  But, since this is an overall obscuring key, it may be used in other places later on.